subsys/net/ip/net_context.c

```diff
@@ -332,9 +353,34 @@ static uint16_t find_available_port(struct net_context *context,
 	uint16_t local_port;
 
 	do {
-		local_port = sys_rand16_get() | 0x8000;
+		if (IS_ENABLED(CONFIG_NET_CONTEXT_CLAMP_PORT_RANGE)) {
+			uint16_t upper, lower;
+
+			upper = COND_CODE_1(CONFIG_NET_CONTEXT_CLAMP_PORT_RANGE,
+					    (context->options.port_range >> 16),
+					    (0));
+			lower = COND_CODE_1(CONFIG_NET_CONTEXT_CLAMP_PORT_RANGE,
+					    (context->options.port_range & 0xffff),
+					    (0));
+
+			/* This works the same way as in Linux. If either port
+			 * range is 0, then we use random port. If both are set,
+			 * then we use the range. Also make sure that upper is
+			 * greater than lower.
+			 */
+			if (upper == 0 || lower == 0 || upper < lower) {
```

Comment:
If upper is supposed to be greater than lower, then we should check for `upper <= lower` here, right?

Reply:
But then we would not be able to have a range of one port number. That would probably not be very practical but I suppose it could be done.

Reply:
Good point, I think we need to stick with what the doc says as it is copied from Linux. I will fix the checks.